src/event.js

```diff
@@ -522,11 +522,12 @@ function leverageNative( el, type, isSetup ) {
 			if ( ( event.isTrigger & 1 ) && this[ type ] ) {
 
 				// Interrupt processing of the outer synthetic .trigger()ed event
-				if ( !saved ) {
+				// Detect `saved` of shape `{ value }` and `false`.
+				if ( !saved.length ) {
 
 					// Store arguments for use when handling the inner native event
-					// There will always be at least one argument (an event object), so this array
-					// will not be confused with a leftover capture object.
+					// There will always be at least one argument (an event object),
+					// so this array will not be confused with a leftover capture object.
```

Comment:
We should experiment with checking `isArray(saved)` rather than `saved.length` (as both point replacements in the `if` conditions and as a dedicated variable like `savedIsArgs = Array.isArray(saved)`). If there's no size increase, then this commentary would be unnecessary.

Reply:
Changing `saved.length` checks to `Array.isArray( saved )` results in a size increase:
```
   raw     gz Filename
   +16     +7 dist/jquery.min.js
   +16     +1 dist/jquery.slim.min.js
   +16     +4 dist-module/jquery.module.min.js
   +16      0 dist-module/jquery.slim.module.min.js
```